### ISSUE TITLE
File /lib/libml/dmlc/xgboost4j.so was not found inside JAR.

### ISSUE THREAD

**Issue by martijndoornik** [dmlc/xgboost#1471]
I'm experimenting with several learning machines, and I encountered XGBoost as one. Trying to run it as a regressor in my own code, it crashes, stating that the native file is not found. Full stack trace: 

Aug 16, 2016 9:56:31 AM ml.dmlc.xgboost4j.java.NativeLibLoader smartLoad
SEVERE: failed to load library from both native path and jar
Aug 16, 2016 9:56:31 AM ml.dmlc.xgboost4j.java.DMatrix <clinit>
SEVERE: load native library failed.
Aug 16, 2016 9:56:31 AM ml.dmlc.xgboost4j.java.DMatrix <clinit>
SEVERE: java.io.FileNotFoundException: File /lib/libml/dmlc/xgboost4j.so was not found inside JAR.
Exception in thread "main" java.lang.UnsatisfiedLinkError: ml.dmlc.xgboost4j.java.XGBoostJNI.XGDMatrixCreateFromMat([FIIF[J)I
	at ml.dmlc.xgboost4j.java.XGBoostJNI.XGDMatrixCreateFromMat(Native Method)
	at ml.dmlc.xgboost4j.java.DMatrix.<init>(DMatrix.java:117)
	at com.experty.gboost.GradientBoosting.predict(GradientBoosting.java:40)

But in the downloaded directory, I cannot find this file either. Is there somewhere where I can find this file. My Google Fu has resulted in nothing. 

Thanks in advance.

**Comment by universewill** [comment]
I also got this problem. How do you solve it?

**Comment by martijndoornik** [comment]
I solved it by first downloading the [dmlc-core](https://github.com/dmlc/dmlc-core/tree/d8d4dccb8bd89af9f57aa19e7523f0b755563ca4) and [rabit](https://github.com/dmlc/rabit/tree/2dd7476ad7e4afe97eb579d39aab226ff74400a6) packages and placing them in the corresponding folders in the xgboost folder, as they do not come with the download of xgboost itself. Then, I used the "make" command in the jvm-package folder. That gives a jar-file, which I added to the project and there we go: done. 
Hope it helps
